File: src/main/java/com/IceCreamQAQ/Yu/controller/ActionInvoker.java
package com.IceCreamQAQ.Yu.controller;

import com.IceCreamQAQ.Yu.controller.route.RouteInvoker;
import lombok.Data;

@Deprecated
@Data
public abstract class ActionInvoker implements RouteInvoker {

    protected MethodInvoker[] befores;
    protected MethodInvoker invoker;

    @Override
    public abstract void invoke(String path, ActionContextBase context) throws Exception;
}

File: src/main/kotlin/com/IceCreamQAQ/Yu/controller/old/ActionInvoker.kt
package com.IceCreamQAQ.Yu.controller.old

interface ActionInvoker : Router {
    val befores: Array<MethodInvoker>
    val invoker: MethodInvoker
    val afters: Array<MethodInvoker>
    val catchs: Array<CatchInvoker>
}
